docs/structure-of-a-contract.rst

```diff
@@ -42,6 +42,43 @@ Functions are the executable units of code within a contract.
 and have different levels of visibility (:ref:`visibility-and-getters`)
 towards other contracts. Functions must be decorated with either @public or @private.
 
+Default function
+----------------
+
+A contract can also have a default function, which is executed on a call to the contract if no other functions match the given function identifier (or if none was supplied at all, such as through someone sending it Eth). It is the same construct as fallback functions `in Solidity <https://solidity.readthedocs.io/en/latest/contracts.html?highlight=fallback#fallback-function>`_. 
+
+This function is always named `__default__` and must be annotated with `@public`. It cannot have arguments and cannot return anything.
+
+If the function is annotated as `@payable`, this function is executed whenever the contract is sent Ether (without data). This is why the default function cannot accept arguments and return values - it is a design decision of Ethereum to make no differentiation between sending ether to a contract or a user address.
+
+**Example:**
+
+::
+
+    Payment: event({amount: int128, from: indexed(address)})
+    
+    @public
+    @payable
+    def __default__():
+        log.Payment(msg.value, msg.sender)
+
+
+Considerations
+~~~~~~~~~~~~~~
+
+Unlike Solidity, Vyper generates a default function is one isn't found, in the form of a REVERT call. Note that this still `generates an exception <https://github.com/ethereum/wiki/wiki/Subtleties>`_ and thus will not succeed in receiving funds. 
+
+Ethereum specifies that the operations will be rolled back if the contract runs out of gas in execution. ``send`` calls to the contract come with a free stipend of 2300 gas, which does not leave much room to perform other operations except basic logging. **However**, if the sender includes a higher gas amount through a ``call`` instead of ``send``, then more complex functionality can be run.
+
+The following operations will consume more gas than the 2300 gas stipend:
+
+- Writing to storage
+- Creating a contract
+- Calling an external function which consumes a large amount of gas
+- Sending Ether
+
+Since global variables are still accessible, you can pass data to the default function through the raw byte buffer provided to EVM's CALL (``msg.data``).
```

Comment:
I think accessing `msg.data` is discouraged behavior. Please mention that, or skip this section. You could perhaps mention accessing `msg.sender` instead.

Reply:
+1 a block mentioning that when using `msg.data` care should be taking.